t/opendir.t

```diff
@@ -84,5 +84,19 @@ is(
     'Symlink appears in directory content'
 );
 
+{
+    my $d1 = Test::MockFile->dir('/foo2/bar');
+    my $d2 = Test::MockFile->dir('/foo2');
+    mkdir $d1->path();
+    mkdir $d2->path();
```

Comment:
should not the `mkdir` order be reverted?

Reply:
The order has nothing to do with the bug. The bug is that a file within an inner directory is shown instead of just the inner directory.